src/components/FormAlertWithSubmitButton.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import {StyleProp, View, ViewStyle} from 'react-native';
+import useThemeStyles from '@styles/useThemeStyles';
+import Button from './Button';
+import FormAlertWrapper from './FormAlertWrapper';
+
+type FormAlertWithSubmitButtonProps = {
+    /** Error message to display above button */
+    message?: string;
+
+    /** Whether the button is disabled */
+    isDisabled?: boolean;
+
+    /** Whether message is in html format */
+    isMessageHtml?: boolean;
+
+    /** Styles for container element */
+    containerStyles?: StyleProp<ViewStyle>;
+
+    /** Is the button in a loading state */
+    isLoading?: boolean;
+
+    /** Callback fired when the "fix the errors" link is pressed */
+    onFixTheErrorsLinkPressed?: () => void;
+
+    /** Submit function */
+    onSubmit: () => void;
+
+    /** Should the button be enabled when offline */
+    enabledWhenOffline?: boolean;
+
+    /** Disable press on enter for submit button */
+    disablePressOnEnter?: boolean;
+
+    /** Whether the form submit action is dangerous */
+    isSubmitActionDangerous?: boolean;
+
+    /** Custom content to display in the footer after submit button */
+    footerContent?: React.ReactNode;
+
+    /** Styles for the button */
+    buttonStyles?: StyleProp<ViewStyle>;
+
+    /** Whether to show the alert text */
+    isAlertVisible: boolean;
+
+    /** Text for the button */
+    buttonText: string;
+};
+
+function FormAlertWithSubmitButton({
+    message = '',
+    isDisabled = false,
+    isMessageHtml = false,
+    containerStyles,
+    isLoading = false,
+    onFixTheErrorsLinkPressed = () => {},
+    enabledWhenOffline = false,
+    disablePressOnEnter = false,
+    isSubmitActionDangerous = false,
+    footerContent = null,
+    buttonStyles,
+    buttonText,
+    isAlertVisible,
+    onSubmit,
+}: FormAlertWithSubmitButtonProps) {
+    const styles = useThemeStyles();
+    const style = [!footerContent ? {} : styles.mb3, buttonStyles];
+
+    return (
+        <FormAlertWrapper
+            containerStyles={[styles.mh5, styles.mb5, styles.justifyContentEnd, containerStyles]}
+            isAlertVisible={isAlertVisible}
+            isMessageHtml={isMessageHtml}
+            message={message}
+            onFixTheErrorsLinkPressed={onFixTheErrorsLinkPressed}
+        >
+            {(isOffline: boolean) => (
+                <View>
+                    {isOffline && !enabledWhenOffline ? (
```

Comment:
I know that this is out of scope with this PR purpose but there's code duplication in this ternary and should be addressed

Reply:
I see your point but during TS migration we should not do refactors which are not related to migration, maybe you can create a seperate issue for that?